client/dashboard/sites/settings-sftp-ssh/clipboard-input-control.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+	// eslint-disable-next-line wpcalypso/no-unsafe-wp-apis
+	__experimentalInputControl as InputControl,
+	// eslint-disable-next-line wpcalypso/no-unsafe-wp-apis
+	__experimentalInputControlSuffixWrapper as InputControlSuffixWrapper,
+	Button,
+} from '@wordpress/components';
+import { sprintf, __ } from '@wordpress/i18n';
+import { copySmall } from '@wordpress/icons';
+import React, { useState, useEffect } from 'react';
+
+export default function ClipboardInputControl( {
+	onCopy,
+	...props
+}: React.ComponentProps< typeof InputControl > & { onCopy?: ( label: string ) => void } ) {
+	const [ isCopied, setCopied ] = useState( false );
+
+	const handleCopy = () => {
+		if ( props.value ) {
+			navigator.clipboard.writeText( props.value );
+			setCopied( true );
+			onCopy?.( props.label );
+		}
+	};
+
+	// toggle the `isCopied` flag back to `false` after 4 seconds
+	useEffect( () => {
+		if ( isCopied ) {
+			const timerId = window.setTimeout( () => setCopied( false ), 4000 );
+			return () => window.clearTimeout( timerId );
+		}
+	}, [ isCopied ] );
+
+	return (
+		<InputControl
+			{ ...props }
+			suffix={
+				<InputControlSuffixWrapper variant="control">
+					<Button
+						size="small"
+						icon={ copySmall }
+						label={
+							isCopied
+								? __( 'Copied' )
+								: sprintf(
+										/* translators: %s is the field to copy */
+										__( 'Copy %s' ),
+										props.label
```

Comment:
Is it possible for use to downcast the field name?

At the moment we show `Copy Username`, but to be consistent with our copy guidelines this should really be sentence case, so `Copy username`. We’d need to watch out for cases where the field name is an acronym though, like `Copy URL` which should remain all-caps.

Reply:
The same applies to the Snackbar we show when the value is copied.

<img width="319" alt="Screenshot 2025-06-03 at 11 51 21" src="https://github.com/user-attachments/assets/fd53fb1d-1c92-4a15-a1c3-931cc971bd24" />

This should really be:

> Copied connection command to clipboard.

Reply:
Maybe for this first iteration, if it's too convoluted, we can just show `Copied!`?

(Trying to unblock things here)

Reply:
`Copied!` is a bit too generic for the snackbar. If you copy a few different fields in quick successions, you'll end up with a bunch of identical snackbars.

Reply:
I think perhaps this isn't the best use of `sprintf` 🤔 E.g. what if "Connection command" is a masculine noun and "Username" is feminine? Or perhaps when you're referring to "connection command" in the past tense (e.g. "copied connection command") it should be different. There may be no good way to to translate them generically.

I think we should just have totally separate strings for `'Connection command'`, `'Copied connection command to clipboard.'`, and `'Copy connection command'`. They can all be translated separately.

Reply:
I'm not entirely sure, but maybe we can just use lowercase for the label text, since it will be transformed to uppercase by CSS anyway. For example:
* URL → url
* Username → username

This might help keep things consistent and easier to manage in the code.

Reply:
Would URL to uppercase in the tooltip/snackbar though? I don't think we have the CSS rule there.

Reply:
Sorry, the example was incorrect 🤦‍♂️ We'll keep URL in uppercase.